### PR TITLE
update for discovery selectors, cluster-wide-access and namespace watching

### LINKED ISSUE #6054: update kiali.io docs for discovery selector, cluster-wide-access and namespace watching
We recently added support for Istio discoverySelectors, a new cluster-wide-access feature, and operator namespace watching:

- https://github.com/kiali/kiali/pull/6027
- https://github.com/kiali/kiali-operator/pull/628

We need kiali.io docs for this. And we need to announce this in release notes for 1.67.

### PR DESCRIPTION
**Pull request by jmazzitelli**
fixes: https://github.com/kiali/kiali/issues/6054

Netlify links:

* https://deploy-preview-649--kiali.netlify.app/docs/configuration/namespace-management/
* https://deploy-preview-649--kiali.netlify.app/docs/faq/general/#why-isnt-my-namespace-in-the-namespace-selection-dropdown
* https://deploy-preview-649--kiali.netlify.app/docs/faq/installation/#operator-fails-due-to-cannot-list-resource-clusterroles-error

### CODE REVIEW

File: content/en/docs/Configuration/namespace-management.md [hunk 1]
```diff
@@ -10,15 +10,22 @@ The default Kiali installation (as mentioned in the [Installation guide]({{< ref
...
-of the ones you want, excluding the ones you don't want, or filtering by a
-label selector. You can use a combination of these options.
+of the ones you want, excluding the ones you don't want. You can also filter namespaces by using [Istio's discovery selectors](https://istio.io/latest/docs/reference/config/istio.mesh.v1alpha1/#MeshConfig). You can use a combination of these options together.
+
+{{% alert color="warning" %}}
+As of Kiali 1.67, the following settings are deprecated:
+* api.namespaces.exclude
+* api.namespaces.include
+* api.namespaces.label_selector_exclude
+* api.namespaces.label_selector_include
```
Comment (jshaughn) [inline content/en/docs/Configuration/namespace-management.md:20]:
```suggestion
* api.namespaces.label_selector_include

It is recommended to instead use Istio Discovery Selectors to limit the namespaces in the mesh.
```

File: content/en/docs/Configuration/namespace-management.md [hunk 2]
```diff
@@ -34,31 +41,31 @@ spec:
     - mycorp_.*
 ```
 
-The operator will create a Role for each accessible namespace. Each Role will be assigned to the Kiali Service Account thus providing Kiali access to those namespaces.
+By default, the operator will create a Role for each accessible namespace. Each Role will be assigned to the Kiali Service Account thus providing Kiali access to those namespaces.
```
Comment (jshaughn) [inline content/en/docs/Configuration/namespace-management.md]: This isn't totally true is it?  The behavior depends on how CWA is set.  The default value for CWA depends on whether AN specifies an explicit namespace list.  If CWA is true then you get a cluster-role, if false you get namespace roles.

Also, maybe change, "providing Kiali access to" to "providing Kiali permissions to" ?

File: content/en/docs/Configuration/namespace-management.md [hunk 3]
```diff
@@ -34,31 +41,31 @@ spec:
...
+
+{{% alert color="info" %}}
+There is an additional setting `deployment.cluster_wide_access` that may be relevant for certain Kiali installations. Because Kiali server utilizes Kubernetes watches to watch all namespaces in `deployment.accessible_namespaces`, for very large clusters this may cause performance issues. In cases like this, to increase performance you can set `deployment.cluster_wide_access` to `true` when specifying a large list of namespaces in `accessible_namespaces`. When you do this, the Kiali Server will be given access to the entire cluster and thus it can use a single cluster watch which increases performance and efficiency. However, you must be aware that when you do this, the Kiali Server will be granted access to the cluster via a ClusterRole - individual Roles will not be created per namespace. The `deployment.accessible_namespaces` will still be used to determine which namespaces to make available to users.
+{{% /alert %}}
```
Comment (jshaughn) [inline content/en/docs/Configuration/namespace-management.md]: I don't think you should assume that cluster-wide access is just for very large installations.  I think we should just tell users the behavior and that there are performance benefits to CWA true in exchange for the less explicit access.  They can decide what they want to do.

File: content/en/docs/Configuration/namespace-management.md [hunk 4]
```diff
@@ -67,17 +74,29 @@ set must be matched by only one Kiali CR. Regular expressions must not have
...
+
+{{% alert color="info" %}}
+You must make sure you configure `accessible_namespaces` with a list of namespaces that are equal to, or are a superset of, the namespaces that result from evaluating the Istio discovery selectors.
+{{% /alert %}}
+
 ## Included Namespaces
 
-When `accessible_namespaces` is set to `["**"]` you can still limit the namespaces a user will see in Kiali. The `api.namespaces.include` setting allows you to configure the subset of namespaces that Kiali will show the user. This list is specified in a way similar to `accessible_namespaces`, as a list of namespaces that can include regex patterns. The difference is that this list is processed by the server, not the operator, each time the list of namespaces needs to be obtained by Kiali. This means it will handle namespaces that exist when Kiali is installed, or namespaces created later.
+{{% alert color="warning" %}}
+This feature is deprecated as of 1.67.
```
Comment (jshaughn) [inline content/en/docs/Configuration/namespace-management.md]:
```suggestion
This feature is deprecated as of 1.67.  It is recommended to instead use Istio Discovery Selectors.
```

File: content/en/docs/Configuration/namespace-management.md [hunk 5]
```diff
@@ -102,7 +120,11 @@ Note that this setting is merely a filter and does not provide security in any w
 
 ## Excluded Namespaces
 
-The `api.namespaces.exclude` setting configures namespaces to exclude from being shown to the user. This setting can be used both when `accessible_namespaces` is set to `["**"]` or set to an explicit list of namespaces. It can also be used regardless of whether `api.namespaces.include` is defined. The exclude filter has precedence - if a namespace matches any regex pattern defined for `api.namespaces.exclude`, it will not be shown in Kiali. Like `api.namespaces.include`, the exclude setting is only a filter and does not provide security in any way.
+{{% alert color="warning" %}}
+This feature is deprecated as of 1.67.
```
Comment (jshaughn) [inline content/en/docs/Configuration/namespace-management.md]:
```suggestion
This feature is deprecated as of 1.67.  It is recommended to instead use Istio Discovery Selectors.
```

File: content/en/docs/Configuration/namespace-management.md [hunk 6]
```diff
@@ -124,6 +146,10 @@ You cannot exclude the control plane namespace. If you specify it in `api.namesp
 
 ## Namespace Selectors
 
+{{% alert color="warning" %}}
+This feature is deprecated as of 1.67.
```
Comment (jshaughn) [inline content/en/docs/Configuration/namespace-management.md]:
```suggestion
This feature is deprecated as of 1.67.  It is recommended to instead use Istio Discovery Selectors.
```

**Review by jshaughn — COMMENTED** [review]
Looks good, I like that `**` is no longer something users have to understand.  Some comments/possible changes...

File: content/en/docs/Configuration/namespace-management.md (hunk 2 above)
Comment (jmazzitelli) [reply content/en/docs/Configuration/namespace-management.md]: will re-word this and add "permissions to" blurb. Stay tuned for next commit.

File: content/en/docs/Configuration/namespace-management.md (hunk 3 above)
Comment (jmazzitelli) [reply content/en/docs/Configuration/namespace-management.md]: will take that part out about very large installs. stay tuned for next commit

File: content/en/docs/Configuration/namespace-management.md (hunk 4 above)
Comment (jmazzitelli) [reply content/en/docs/Configuration/namespace-management.md]: done

File: content/en/docs/Configuration/namespace-management.md (hunk 5 above)
Comment (jmazzitelli) [reply content/en/docs/Configuration/namespace-management.md]: done

File: content/en/docs/Configuration/namespace-management.md (hunk 6 above)
Comment (jmazzitelli) [reply content/en/docs/Configuration/namespace-management.md]: done

**Comment by jmazzitelli** [comment]
latest commit has changes to reflect suggestions. ready for review again

**Review by jshaughn — APPROVED** [review]
Looks good, we can tweak more given any user feedback/questions.

**Pull request merged**